### ISSUE TITLE
bond interface stuck in "configuring" without `LinkLocalAddressing=no`

### ISSUE THREAD

**Issue by LaserEyess** [systemd/systemd#24717]
### systemd version the issue has been seen with

systemd 251 (251.4-1-arch)

### Used distribution

Arch Linux

### Linux kernel version used

5.15.68-1-lts

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

networkd realizes getting an ipv6 via link local addressing is not possible

### Unexpected behaviour you saw

With a bonded interface on a network without any ipv6 at the router level, it seems that networkd cannot figure this out and considers the link as permanently "configuring", which causes `systemd-networkd-wait-online` to fail. 

This seems very similar to #8686 

### Steps to reproduce the problem

1. turn off ipv6 at the router level (different machine)
2. create a `.network` file with a static ipv4 for a bond interface
3. do *not* add `LinkLocalAddressing=no` to your `.network` file
4. interface is permanently stuck in "configuring"

I don't know if a bonded interface is necessary, but I can reproduce it with one.

Log files with `LinkLocalAddressing` as the default value: https://0x0.st/oOFY.logs
Log files with `LinkLocalAddressing=no`: https://0x0.st/oOF3.logs

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
# See above for logs

# 10-bond0.netdev
[NetDev]
Name=bond0
Kind=bond
MACAddress = d0:50:99:c1:f0:89

[Bond]
Mode=balance-xor
TransmitHashPolicy=layer2+3

####

# 10-bond0.network
[Match]
Name=bond0

[Network]
# Bug see
# https://github.com/systemd/systemd/issues/8686
LinkLocalAddressing=no
Address=10.0.1.4/24
Gateway=10.0.1.1

DNS=10.0.1.1
Domains=example.domain
DNSDefaultRoute=true

NTP=10.0.1.1

[Link]
RequiredForOnline=degraded-carrier

# 20-eth.network
[Match]
PermanentMACAddress=d0:50:99:f8:05:53 d0:50:99:f8:05:52

[Network]
Bond=bond0
ConfigureWithoutCarrier=true

[Link]
RequiredForOnline=false
bebe@charlotte network % cat 20-eth.network
[Match]
PermanentMACAddress=d0:50:99:f8:05:53 d0:50:99:f8:05:52

[Network]
Bond=bond0
ConfigureWithoutCarrier=true

[Link]
RequiredForOnline=false
```

**Comment by yuwata** [comment]
Sorry for late response. The following log messages related to the issue. I will take a look.
```
bond0: NDISC: Error sending Router Solicitation: No buffer space available
bond0: NDISC: Stopping IPv6 Router Solicitation client
```

**Comment by yuwata** [comment]
Fix is waiting in #24957. If possible, please test the PR. Thank you.

**Comment by LaserEyess** [comment]
Apologies for the late reply, but this is now in 251.6 and it does indeed fix my issue. Thanks!
